webpack/components/SelectAllCheckbox/index.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Dropdown, DropdownToggle, DropdownToggleCheckbox,
+  DropdownItem } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { noop } from 'foremanReact/common/helpers';
+
+import { pluralize } from '../../utils/helpers';
+import './SelectAllCheckbox.scss';
+
+const SelectAllCheckbox = ({
+  selectAll,
+  selectNone,
+  selectPage,
+  selectedCount,
+  modelName,
+  modelNamePlural,
+  areAllRowsOnPageSelected,
+  areAllRowsSelected,
+}) => {
+  const [isSelectAllChecked, setSelectAllChecked] = useState(null);
+  const [isSelectAllDropdownOpen, setSelectAllDropdownOpen] = useState(false);
+
+  const onSelectAllCheckboxChange = () => {
+    if (isSelectAllChecked === false) {
+      return selectPage();
+    }
+    return selectNone();
+  };
+  const onSelectAllDropdownToggle = () => setSelectAllDropdownOpen(isOpen => !isOpen);
+  const pluralizedModel = pluralize(
+    selectedCount,
+    modelName,
+    modelNamePlural,
+  );
+
+  const handleSelectAll = () => {
+    selectAll();
+    setSelectAllDropdownOpen(false);
+  };
+  const handleSelectPage = () => {
+    selectPage();
+    setSelectAllDropdownOpen(false);
+  };
+  const handleSelectNone = () => {
+    selectNone();
+    setSelectAllDropdownOpen(false);
+  };
+
+  useEffect(() => {
+    let newCheckedState;
+    if (selectedCount === 0) newCheckedState = false;
+    if (selectedCount > 0) newCheckedState = null; // null is partially-checked state
+    if (areAllRowsSelected) newCheckedState = true;
+    setSelectAllChecked(newCheckedState);
+  }, [selectedCount, areAllRowsSelected]);
+
+  const selectAllDropdownItems = [
+    <DropdownItem key="select-all" component="button" isDisabled onClick={handleSelectAll}>
+      {__('Select all')}
+    </DropdownItem>,
+    <DropdownItem key="select-page" component="button" isDisabled={areAllRowsOnPageSelected} onClick={handleSelectPage}>
```

Comment:
There was talk of removing items instead of disabling them when not relevant, not sure if Maria had weighed in on that for the dropdown items?

Reply:
I think you're looking for @MariSvirik :)